Client/NotificationManager.swift

```diff
@@ -0,0 +1,152 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import UserNotifications
+import Shared
+
+protocol UserNotificationCenterProtocol {
+    func getNotificationSettings(completionHandler: @escaping (UNNotificationSettings) -> Void)
+    func requestAuthorization(options: UNAuthorizationOptions,
+                              completionHandler: @escaping (Bool, Error?) -> Void)
+    func add(_ request: UNNotificationRequest, withCompletionHandler completionHandler: ((Error?) -> Void)?)
+    func getPendingNotificationRequests(completionHandler: @escaping ([UNNotificationRequest]) -> Void)
+    func removePendingNotificationRequests(withIdentifiers identifiers: [String])
+    func removeAllPendingNotificationRequests()
+    func getDeliveredNotifications(completionHandler: @escaping ([UNNotification]) -> Void)
+    func removeDeliveredNotifications(withIdentifiers identifiers: [String])
+    func removeAllDeliveredNotifications()
+}
+
+extension UNUserNotificationCenter: UserNotificationCenterProtocol {
+}
```

Comment:
nit: any particular reason for this empty extension 🤔

Reply:
The notification manager has the variable `center` that adheres to this protocol. This way we can mock it for tests. If UNUserNotificationCenter doesn't implement the protocol we wouldn't be able to assign an instance to `center`.

Reply:
It would be good to add a note. As a quick note would be beneficial for folks trying to understand why is there an empty extension.